include/circt-c/Dialect/FIRRTL.h

```diff
@@ -12,13 +12,119 @@
 #define CIRCT_C_DIALECT_FIRRTL_H
 
 #include "mlir-c/IR.h"
+#include "mlir-c/Support.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+//===----------------------------------------------------------------------===//
+// Dialect API.
+//===----------------------------------------------------------------------===//
+
 MLIR_DECLARE_CAPI_DIALECT_REGISTRATION(FIRRTL, firrtl);
 
+//===----------------------------------------------------------------------===//
+// Type API.
+//===----------------------------------------------------------------------===//
+
+/// Return 'true' if this is any FIRRTL type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsFIRRTLType(MlirType type);
+
+/// Return 'true' if this is a FIRRTL ground type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsGround(MlirType type);
+
+/// Return 'true' if this is a FIRRTL const type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsConst(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Clock type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsClock(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Reset type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsReset(MlirType type);
+
+/// Return 'true' if this is a FIRRTL AsyncReset type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsAsyncReset(MlirType type);
+
+/// Return 'true' if this is a FIRRTL SInt type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsSInt(MlirType type);
+
+/// Return 'true' if this is a FIRRTL UInt type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsUInt(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Analog type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsAnalog(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Bundle type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsBundle(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Vector type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsFVector(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Reference type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsRef(MlirType type);
+
+/// Return 'true' if this is a FIRRTL OpenBundle type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsOpenBundle(MlirType type);
+
+/// Return 'true' if this is a FIRRTL OpenVector type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsOpenVector(MlirType type);
```

Comment:
I don't know much about the Open types, which seems to be used internally in CIRCT only. (`FIRParser.cpp` does not use it)

Is it necessary to expose APIs for Open types?